hi-ml-azure/src/health_azure/himl.py

```diff
@@ -708,15 +708,22 @@ def create_v2_inputs(ml_client: MLClient, input_datasets: List[DatasetConfig]) -
         input_name = f"INPUT_{i}"
         version = input_dataset.version or 1
         data_asset: Data = ml_client.data.get(input_dataset.name, version=str(version))
-        data_path = data_asset.id or ""
-        # Note that there are alternative formats that the input path can take, such as:
-        # v1_datastore_path = f"azureml://datastores/{input_dataset.datastore}/paths/<path_to_dataset>"
-        # v2_dataset_path = f"azureml:{input_dataset.name}:1"
-
-        inputs[input_name] = Input(  # type: ignore
-            type=AssetTypes.URI_FOLDER,
+        if not data_asset.path:
+            raise ValueError(f"Data asset {data_asset.id} has no path.")
+        # Some mismatches with the documentation here:
```

Comment:
I think the comments here are so specialized to the code that they don't fit into the docstring. This is more of an explanation "why is the code the way it is" - because it's far from obvious.